utils: remove swiftinterfaces from the build image

compnerd opened the pull request:
These cannot be reserialized due to a bug in the serialization logic where we do not inject the VFS Ovelay for the system modules.

Bug: #79839

compnerd commented:
Please test with following PRs:
https://github.com/swiftlang/swift-installer-scripts/pull/400

@swift-ci please smoke test

compnerd commented:
Please test with following PRs:
https://github.com/swiftlang/swift-installer-scripts/pull/400

@swift-ci please build toolchain Windows platform

compnerd commented:
CC: @xymus

weliveindetail commented on utils/build.ps1:
```diff
@@ -3207,6 +3207,10 @@ if (-not $SkipBuild) {
     Invoke-BuildStep Build-Runtime Windows $Arch
+
+    # FIXME(#79839)
+    Get-ChildItem -Path "$($Arch.SDKRoot)\usr\lib\swift\android" -Filter "*.swiftinterface" -Recurse
```
Was this supposed to look into `\usr\lib\swift\windows`? I think we can skip it altogether as the Windows SDKs don't reproduce the error.

compnerd replied:
Yes, it was supposed to look into Windows. I've tested it locally and definitely see the issues on Windows locally :)

weliveindetail replied:
Interesting, doesn't happen for me locally

weliveindetail commented on utils/build.ps1:
```diff
@@ -3232,6 +3236,10 @@ if (-not $SkipBuild) {
     Invoke-BuildStep Build-Runtime Android $Arch
+
+    # FIXME(#79839)
+    Get-ChildItem -Path "$($Arch.SDKRoot)\usr\lib\swift\android" -Filter "*.swiftinterface" -Recurse
```
I think it should be:
```suggestion
    Get-ChildItem -Path "$($Arch.SDKInstallRoot)\usr\lib\swift\android" -Filter "*.swiftinterface" -Recurse | Remove-Item -Force
```

weliveindetail reviewed (approved):
Agree, that this workaround is the best way forward right now. Test on Windows failed with `property 'SDKRoot' cannot be found`.

compnerd committed:
utils: remove swiftinterfaces from the build image

These cannot be re-serialized due to a bug in the serialization logic
where we do not inject the VFS overlay for the system modules.

Bug: #79839

compnerd commented:
Please test with following PRs:
https://github.com/swiftlang/swift-installer-scripts/pull/400

@swift-ci please build toolchain Windows platform

compnerd commented:
Please test with following PRs:
https://github.com/swiftlang/swift-installer-scripts/pull/400

@swift-ci please smoke test

compnerd commented:
Please test with following PRs:
https://github.com/swiftlang/swift-installer-scripts/pull/400

@swift-ci please smoke test

weliveindetail reviewed (commented):
I tested this in https://github.com/swiftlang/swift/pull/79185 and this specific aspect is working in CI with the tiny fix below.

compnerd commented:
@swift-ci please smoke test macOS platform

compnerd commented:
Please test with following PRs:
https://github.com/swiftlang/swift-installer-scripts/pull/400

@swift-ci please test Windows platform

compnerd commented:
Please test with following PRs:
https://github.com/swiftlang/swift-installer-scripts/pull/400

@swift-ci please build toolchain Windows platform

xymus commented:
It looks like this will correctly work around the change from #79588. Relying only on binary swiftmodules for a local build should be fine, but I would caution against it for distributed SDKs and toolchains.

For more context, we're very slowly moving towards using more and more textual swiftinterfaces for distributed modules. Binary swiftmodules are not designed to support changes in dependencies. Notably, a change to imported headers can force the compiler to drop entire decls, that's in the degraded mode as it crashes otherwise. So when building the modules on one machine and deploying on a different one the binary swiftmodules may be unusable, whereas swiftinterfaces are designed to be rebuilt on each machine against the local dependencies.

compnerd commented:
@xymus hmm, the problem is that the re-serialization will fail with module not found errors for underlying clang modules. I suspect that this is due to VFS overlays not being applied when re-serialising the modules.

weliveindetail commented:
https://github.com/swiftlang/swift-installer-scripts/pull/400

@swift-ci please test Windows

weliveindetail commented:
The Windows test failed, because PR testing is blocked right now (see https://github.com/swiftlang/swift-build/pull/300).

We need this patch (or a improved variant of it) to unblock the Windows x64 mainline bot: https://ci-external.swift.org/job/swift-main-windows-toolchain/

It would be great to get this bot back to green before I land my PR for the Android Swift Runtime tests. One of the PR-test runs over there demonstrated the effectiveness of the patch from this PR in combination with the installer-scripts patch linked above: https://ci-external.swift.org/job/swift-PR-windows/37567/

What is the state here? Can we land this once we get PR-testing unblocked?

compnerd commented:
I think that we can merge it for unblocking the nightlies but would need to restore the swiftinterfaces soon. I think that https://github.com/swiftlang/swift/pull/79926 might be the key to that.

weliveindetail commented:
https://github.com/swiftlang/swift-installer-scripts/pull/400
https://github.com/swiftlang/swift-build/pull/300
@swift-ci please test Windows

hjyamauchi commented:
I checked that https://github.com/swiftlang/swift/pull/79926 does **not** fix this particular issue. It feels like we need some  changes to similar to 79926 to fix this issue so far but I haven't figured out yet. I'm on it.
